components/org.wso2.carbon.identity.conditional.auth.functions.session/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/session/function/IsWithinSessionLimitFunction.java

```diff
@@ -0,0 +1,169 @@
+import org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.conditional.auth.functions.session.util.SessionValidationConstants;
+import org.wso2.carbon.identity.conditional.auth.functions.session.util.SessionValidationUtil;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Represents javascript function provided in conditional authentication to check if the given user has valid number of
+ * sessions.
+ * The purpose is to perform dynamic authentication selection based on the active session count.
+ */
+public class IsWithinSessionLimitFunction implements IsValidFunction {
+
+    private static final Log log = LogFactory.getLog(IsWithinSessionLimitFunction.class);
+
+    /**
+     * Method to validate user session a given the authentication context and set of required attributes.
+     *
+     * @param context Authentication context
+     * @param map     Hash map of attributes required for validation
+     * @return boolean value indicating the validation success/failure
+     * @throws AuthenticationFailedException when exception occurred in session retrieving method
+     */
+    @Override
+    public boolean validate(JsAuthenticationContext context, Map<String, String> map)
+            throws AuthenticationFailedException {
+
+        boolean state = false;
+        int sessionLimit = getSessionLimitFromMap(map);
+        AuthenticatedUser authenticatedUser;
+        try{
+            authenticatedUser = context.getWrapped().getLastAuthenticatedUser();
+        } catch (NullPointerException e){
+            throw new AuthenticationFailedException("Failed to identify the Authenticated user from the context",e);
+        }
+        if (authenticatedUser == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Unable to find the authenticated user from the Authentication context.");
+            }
+            throw new AuthenticationFailedException("Unable to find the Authenticated user from previous step");
+        }
+        try {
+            int sessionCount = getActiveSessionCount(authenticatedUser);
+            if (log.isDebugEnabled()) {
+                log.debug("Active session count: " + sessionCount + " and session limit : " + sessionLimit);
+            }
+            if (sessionCount < sessionLimit) {
+                state = true;
+            }
+        } catch (FrameworkException e) {
+            throw new AuthenticationFailedException("Problem occurred retrieving session data for the user " +
+                    authenticatedUser.getUserName(), e);
+        } catch (NumberFormatException e) {
```

Comment:
what's the reason to have this? any integer parsing involved?

Reply:
Integer parsing is involved in getActiveSessionCount() method used in line 83. This is fixed by moving this NFE to getActiveSessionCount() method and throwing FrameworkException when unexpected (non-integer) value is retrieved from HTTP call.